include/grpc++/channel_interface.h

```diff
@@ -57,6 +58,39 @@ class ChannelInterface : public CallHook,
   virtual void* RegisterMethod(const char* method_name) = 0;
   virtual Call CreateCall(const RpcMethod& method, ClientContext* context,
                           CompletionQueue* cq) = 0;
+
+  // Get the current channel state. If the channel is in IDLE and try_to_connect
+  // is set to true, try to connect.
+  virtual grpc_connectivity_state GetState(bool try_to_connect) = 0;
+
+  // Return the tag on cq when the channel state is changed or deadline expires.
+  // GetState needs to called to get the current state.
+  virtual void NotifyOnStateChange(grpc_connectivity_state last_observed,
```

Comment:
(and the same for the other functions - it allows the same model to adapt different time types as we need them)